### ISSUE TITLE
Error: cursor id <id> not found

### ISSUE THREAD

**Issue by deser** [Automattic/mongoose#8249]
**Do you want to request a *feature* or report a *bug*?**
*bug*

**What is the current behavior?**
In latest mongoose ( I tested 5.7.4 and 5.7.5) in my tests I see `Error: cursor id 9058772029368767025 not found` error. When I downgrade to 5.7.1 everything is ok. 

**What is the expected behavior?**
In 5.7.5 my tests doesn't fail 

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
node: v10.16.1
mongodb: 4.2

**Comment by vkarpov15** [comment]
Can you please provide the code that is failing and a stack trace? Without that it's hard for me to guess what the issue might be.

**Comment by AndrewBarba** [comment]
I'm seeing the same issue on 5.7.5. Query looks like:

```javascript
model
      .find(options)
      .read('sp')
      .batchSize(1000)
      .cursor()
      .eachAsync(func, { parallel: 100 })
```

And I'm seeing two variations of the error:

```
MongoError: cursor id 8065450018539360080 is already in use
    at Connection.<anonymous> (/var/task/node_modules/mongodb/lib/core/connection/pool.js:466:61)
    at Connection.emit (events.js:198:13)
    at Connection.EventEmitter.emit (domain.js:448:20)
    at processMessage (/var/task/node_modules/mongodb/lib/core/connection/connection.js:364:10)
    at TLSSocket.<anonymous> (/var/task/node_modules/mongodb/lib/core/connection/connection.js:533:15)
    at TLSSocket.emit (events.js:198:13)
    at TLSSocket.EventEmitter.emit (domain.js:448:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at TLSSocket.Readable.push (_stream_readable.js:224:10)
    at TLSWrap.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

```
MongoError: cursor id 859394183419277556 not found
    at Connection.<anonymous> (/var/task/node_modules/mongodb/lib/core/connection/pool.js:466:61)
    at Connection.emit (events.js:198:13)
    at Connection.EventEmitter.emit (domain.js:448:20)
    at processMessage (/var/task/node_modules/mongodb/lib/core/connection/connection.js:364:10)
    at TLSSocket.<anonymous> (/var/task/node_modules/mongodb/lib/core/connection/connection.js:533:15)
    at TLSSocket.emit (events.js:198:13)
    at TLSSocket.EventEmitter.emit (domain.js:448:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at TLSSocket.Readable.push (_stream_readable.js:224:10)
    at TLSWrap.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by AndrewBarba** [comment]
@vkarpov15 Seems very much related to https://github.com/Automattic/mongoose/issues/8235

**Comment by vkarpov15** [comment]
@AndrewBarba is right, this looks like a duplicate of #8235. Should be fixed in 5.7.6
